src/RedisClusterResourceManager.php

```diff
@@ -0,0 +1,244 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Cache\Storage\Adapter;
+
+use Laminas\Cache\Exception\ExtensionNotLoadedException;
+use Laminas\Cache\Exception\RuntimeException;
+use Laminas\Cache\Storage\Adapter\Exception\RedisRuntimeException;
+use Laminas\Cache\Storage\Plugin\PluginInterface;
+use Laminas\Cache\Storage\Plugin\Serializer;
+use Laminas\Cache\Storage\PluginCapableInterface;
+use RedisCluster as RedisClusterFromExtension;
+use RedisClusterException;
+use ReflectionClass;
+
+use function array_key_exists;
+use function assert;
+use function extension_loaded;
+use function is_int;
+use function strpos;
+
+/**
+ * @psalm-type RedisClusterInfoType = array<string,mixed>&array{redis_version:string}
+ */
+final class RedisClusterResourceManager implements RedisClusterResourceManagerInterface
+{
+    /** @var array<non-empty-string,int>|null */
+    private static $clusterOptionsCache;
+
+    /** @var RedisClusterOptions */
+    private $options;
+
+    /** @psalm-var array<int,mixed> */
+    private $libraryOptions = [];
+
+    public function __construct(RedisClusterOptions $options)
+    {
+        $this->options = $options;
+        if (! extension_loaded('redis')) {
+            throw new ExtensionNotLoadedException('Redis extension is not loaded');
+        }
+    }
+
+    /**
+     * @return array<non-empty-string,int>
+     */
+    private static function getRedisClusterOptions(): array
```

Comment:
@laminas/technical-steering-committee any ideas if something like this can be done in a different way? 
Actually, this wont change very often but some constants are only available when redis extension is compiled with specific options and/or when another extension is compiled (like `igbinary`).